### PR TITLE
Adding Pravetz-16S definition

### PR DESCRIPTION
**Pull request by izne**
Summary
=======
Added Pravetz-16S machine definition

Checklist
=========
* [x] I have discussed this with core contributors already
* [x] This pull request requires changes to the ROM set
  * [x] I have opened a roms pull request - https://github.com/86Box/roms/pull/255

References
==========
[CPU12+.pdf](https://github.com/86Box/86Box/files/14834554/CPU12%2B.pdf)
[Pravetz-16S article in Sandacite.bg](https://sandacite.bg/български-компютър-правец-16s-2/)

### CODE REVIEW

File: src/machine/machine_table.c [hunk 1]
```diff
@@ -1339,6 +1339,45 @@ const machine_t machines[] = {
         .snd_device = NULL,
         .net_device = NULL
     },
+    {
+        .name = "[8088] Pravetz 16S / CPU12 Plus",
+        .internal_name = "pravetz16s",
+        .type = MACHINE_TYPE_8088,
```
[the remark this answers is not in the document]
Comment (izne) [reply src/machine/machine_table.c:1345]: Yes, the NEC V20 was the "turbo" option at 12 MHz and memory up to 1MB.
image

I believe 16S (CPU12-Plus) used a standard XT keyboard, not like the original Pravetz-16 (with the IMKO-4 board).
However the special keyboard device was created for the IMKO-4 one: see "keyboard_pravetz_device".

I use DOSP16.IMG (but did not post it initially as it has been reported it has some ancient virus in it)
image

The other applications I do not have at the moment, but I can ask around.

[the remark this answers is not in the document]
Comment (izne) [reply src/machine/machine_table.c:1345]: DOSP16.IMG can be downloaded from this link: http://old-dos.ru/dl.php?id=1226

https://www.sandacite.com/forum/index.php?PHPSESSID=g3d3371ajfocgek3mnnca4c2ne&topic=10788.msg128010#msg128010

https://poisk-pc.narod.ru/eastpc.htm
http://old-dos.ru/index.php?page=files&mode=files&do=show&id=1050

**Comment by izne** [comment]
> Great to see that!
> 
> Does 86Box support the `C/L` and `C/L Lock` [keys](https://github.com/tmk/tmk_keyboard/wiki/IBM-PC-XT-Keyboard-Protocol#pravetz-16)? 

I believe 16S (CPU12-Plus) used a standard XT keyboard, not like the original Pravetz-16 (with the IMKO-4 board).
Previously, the special keyboard device was created for the IMKO-4 one: see "keyboard_pravetz_device".

> Are there other special features of the Pravetz-16 product line, which may require emulation additions?

The built-in graphics controller (VDC-3) can work in CGA and Hercules modes. Not sure if this is a matter of only a definition now or it requires additions.
image


> Do you happen to have the [Tekst-16 program]

Not at the moment, but looking for it.

>  Or the operating systems [ДОС-16, SPS/DOS-16](https://www.amuseum.bg/katalog/super-11-pravec-16a), ПДОС-16 ([multitasking DOS]

DOSP16.IMG can be downloaded from this link: http://old-dos.ru/dl.php?id=1226

Also have a look here:
https://www.sandacite.com/forum/index.php?PHPSESSID=g3d3371ajfocgek3mnnca4c2ne&topic=10788.msg128010#msg128010
https://poisk-pc.narod.ru/eastpc.htm
http://old-dos.ru/index.php?page=files&mode=files&do=show&id=1050

File: src/machine/machine_table.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (cartifanwlr) [reply src/machine/machine_table.c:1345]: Misc are the Miscellanious machines, the only one we have currently is VPC2007

**Comment by izne** [comment]
Thanks for your interest!


> 1. Are DOS-16 and SPS/DOS-16 two separate operating systems?

SPS is the software publisher that released DOS-16 for the Pravetz-16 line.
I think it is the same OS.

> 2. Is there a SPS/DOS-16 somewhere else? I would like to avoid the one with the virus.

Check this thread https://www.sandacite.com/forum/index.php?topic=17326.0
Quite interesting info and also there is another fresh upload from 360K floppies.

> 3. On the link I provided about DOS-16 are mentioned files KEYBBG.COM, MIKT.COM, IBMT.COM. In the files from the two Russian sites there are multiple KEYBxx.COM (but none is KEYB**BG**.COM), MIK.COM and IBM.COM (but without T at the end). Do you know something about those? Since they deal with keyboard and codepage I would assume exactly here will be the special/customized parts...

No idea here.

> 4. On the link I provided about ПДОС-16 "multitasking" - is this the same as DOSP16.IMG (Правец ДОС-16 / ДОС Правец-16) - or that's a different operating system?

I only can assume it is maybe a different OS. Not sure how to test if the DOSP16.IMG one is multitasking.

> 5. Any further information about "INMOS, съвместима с XENIX, и OS-32" operating systems?

No idea.

**Comment by izne** [comment]
>Please post back here later if you find "Tekst-16"".

Hey @Torinde ! Look what popped out: 
[TEKST-16.zip](https://github.com/86Box/86Box/files/15307033/TEKST-16.zip)

**Pull request merged**